colin-api/openshift/templates/dc.yaml

```diff
@@ -0,0 +1,196 @@
+        role: ${ROLE}
+      annotations:
+        haproxy.router.openshift.io/ip_whitelist: ${ALLOW_LIST}
+    spec:
+      host: ${APPLICATION_DOMAIN}
+      tls:
+        insecureEdgeTerminationPolicy: Redirect
+        termination: edge
+      port:
+        targetPort: ${NAME}-${TAG}-tcp
+      to:
+        kind: Service
+        name: ${NAME}-${TAG}
+        weight: 100
+
+  - kind: DeploymentConfig
+    apiVersion: v1
+    metadata:
+      name: ${NAME}-${TAG}
+      labels:
+        name: ${NAME}
+        environment: ${TAG}
+        role: ${ROLE}
+      annotations:
+        description: Defines how to deploy the application server
+    spec:
+      strategy:
+        rollingParams:
+          intervalSeconds: 1
+          maxSurge: 25%
+          maxUnavailable: 25%
+          timeoutSeconds: 600
+          updatePeriodSeconds: 1
+        type: Rolling
+      triggers:
+        - type: ImageChange
+          imageChangeParams:
+            automatic: true
+            containerNames:
+              - ${NAME}-${TAG}
+            from:
+              kind: ImageStreamTag
+              namespace: ${NAMESPACE}-${IMAGE_NAMESPACE}
+              name: ${NAME}:${TAG}
+      replicas: 1
+      selector:
+        name: ${NAME}
+        environment: ${TAG}
+      template:
+        metadata:
+          name: ${NAME}-${TAG}
+          labels:
+            name: ${NAME}
+            environment: ${TAG}
+            role: ${ROLE}
+        spec:
+          containers:
+            - name: ${NAME}-${TAG}
+              image: ${IMAGE_REGISTRY}/${NAMESPACE}-${IMAGE_NAMESPACE}/${NAME}:${TAG}
+              ports:
+                - containerPort: 8080
+                  protocol: TCP
+              readinessProbe:
+                initialDelaySeconds: 3
+                timeoutSeconds: 30
+                httpGet:
+                  path: /ops/readyz
+                  port: 8080
+              livenessProbe:
+                initialDelaySeconds: 120
+                timeoutSeconds: 30
+                httpGet:
+                  path: /ops/healthz
+                  port: 8080
+
+  - kind: HorizontalPodAutoscaler
+    apiVersion: autoscaling/v1
+    metadata:
+      name: ${NAME}-${TAG}
+      labels:
+        name: ${NAME}
+        environment: ${TAG}
+        role: ${ROLE}
+    spec:
+      scaleTargetRef:
+        kind: DeploymentConfig
+        name: ${NAME}-${TAG}
+      minReplicas: ${{MIN_REPLICAS}}
+      maxReplicas: ${{MAX_REPLICAS}}
+
+parameters:
+  - name: NAME
+    displayName: Name
+    description: The name assigned to all of the OpenShift resources associated to the server instance.
+    required: true
+    value: colin-api
+
+  - name: TAG
+    displayName: Environment TAG name
+    description: The TAG name for this environment, e.g., dev, test, prod
+    value: dev
+    required: true
+
+  - name: ROLE
+    displayName: Role
+    description: Role
+    required: true
+    value: api
+
+  - name: NAMESPACE
```

Comment:
this will need the oracle envvars

Reply:
envvars will manage by 1password. So, I removed all envvars from dc template.